src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs

```diff
@@ -550,6 +583,18 @@ DiagnosticBag analyze(bool strictAnalysis)
                 {
                     bool badRegion = false;
                     walker.Analyze(ref badRegion, result);
+                    if (walker._implicitlyInitializedFieldsOpt is { } implicitlyInitializedFields)
+                    {
+                        Debug.Assert(walker._requireOutParamsAssigned);
```

Comment:
I may need a walkthrough some parts of this change. I didn't understand why we care about `_requireOutParamsAssigned` or what this assertion buys us.
If it is relevant, should `needsDefaultInitialStateForMembers` (in `NullableWalker`) also consider it?

Reply:
Have addressed offline. `!_requireOutParamsAssigned` causes us to skip some of the code paths we need to run in order to produce the full set of implicitly initialized fields. We just avoid building the set at all in this condition and don't want any caller that runs under this condition to try and use the set.